tests/unit/mesh/test_ncmesh.cpp

```diff
@@ -297,6 +297,228 @@ TEST_CASE("pNCMesh PA diagonal",  "[Parallel], [NCMesh]")
+         PCG(A, M, B, X, -1, 500, linear_tol, 0.0);
+#else
+         // 9. If MFEM was compiled with SuiteSparse, use UMFPACK to solve the system.
+         UMFPackSolver umf_solver;
+         umf_solver.Control[UMFPACK_ORDERING] = UMFPACK_ORDERING_METIS;
+         umf_solver.SetOperator(A);
+         umf_solver.Mult(B, X);
+#endif
+
+         a.RecoverFEMSolution(X, b, x);
+         return x.ComputeL2Error(rhs_coef);
+      }();
+
+      auto perror = [&]
+      {
+         // parallel solve
+         ParFiniteElementSpace fes(&pmesh, &fec);
+         ParLinearForm b(&fes);
+
+         ParGridFunction x(&fes);
+         x = 0.0;
+
+         b.AddDomainIntegrator(new DomainLFIntegrator(rhs_coef));
+         b.Assemble();
+
+         ParBilinearForm a(&fes);
+         a.AddDomainIntegrator(new MassIntegrator(one));
+         a.Assemble();
+
+         HypreParMatrix A;
+         Vector B, X;
+         Array<int> empty_tdof_list;
+         a.FormLinearSystem(empty_tdof_list, x, b, A, X, B);
+
+         HypreBoomerAMG amg(A);
+         HyprePCG pcg(A);
+         amg.SetPrintLevel(-1);
+         pcg.SetTol(linear_tol);
+         pcg.SetMaxIter(500);
+         pcg.SetPrintLevel(-1);
+         pcg.SetPreconditioner(amg);
+         pcg.Mult(B, X);
+         a.RecoverFEMSolution(X, b, x);
+         return x.ComputeL2Error(rhs_coef);
+      }();
+
+      constexpr double test_tol = 1e-9;
+      CHECK(std::abs(serror - perror) < test_tol);
+      MPI_Barrier(MPI_COMM_WORLD); // Get our ducks in a row.
+   };
+
+   REQUIRE(smesh.GetNE() == 2);
+   smesh.EnsureNCMesh(true);
+   smesh.Finalize();
+
+   auto partition = std::unique_ptr<int[]>(new int[smesh.GetNE()]);
+   partition[0] = 0;
+   partition[1] = Mpi::WorldSize() > 1 ? 1 : 0;
+
+   auto pmesh = ParMesh(MPI_COMM_WORLD, smesh, partition.get());
+
+   // Construct the NC refined mesh in parallel and serial. Once constructed a
+   // global L2 projected solution should match exactly on each.
+   Array<int> refines, serial_refines(1);
+   if (Mpi::WorldRank() == refining_rank)
+   {
+      refines.Append(0);
+   }
+
+   // Must be called on all ranks as it uses MPI calls internally.
+   // All ranks will use the global element number dictated by rank 0 though.
+   serial_refines[0] = pmesh.GetGlobalElementNum(0);
+   MPI_Bcast(&serial_refines[0], 1, MPI_INT, refining_rank, MPI_COMM_WORLD);
+
+   // Rank 0 refines the parallel mesh, all ranks refine the serial mesh
+   smesh.GeneralRefinement(serial_refines, 1); // nonconformal
+   pmesh.GeneralRefinement(refines, 1); // nonconformal
+
+   REQUIRE(pmesh.GetGlobalNE() == 8 + 1);
+   REQUIRE(smesh.GetNE() == 8 + 1);
+
+   // This is the set of (i,j) that result in face-edge constraints.
+   // These pairs were arrived at by looping over all possible i,j pairs and
+   // checking for the addition of a face-edge constraint.
+   std::vector<std::pair<int,int>> indices{{2,13}, {3,13}, {6,2}, {6,3}};
+
+   // Rank 0 has all but one element.
```

Comment:
```suggestion
   // Rank 0 has all but one element, in parallel with more than 1 rank.
```

Reply:
Went with slightly different wording, but hopefully addressed in https://github.com/mfem/mfem/pull/3587/commits/c790cf942bb329d9b90a6ca1de6b31fccbc1d7d9